src/Data/String/CaseInsensitiveString.purs

```diff
@@ -0,0 +1,22 @@
+module Data.String.CaseInsensitive where
+
+import Prelude
+
+import Data.Newtype (class Newtype)
+import Data.String (toLower)
+
+-- | A newtype for case insensitive string comparisons and ordering.
+newtype CaseInsensitiveString = CaseInsensitiveString String
+
+instance eqCaseInsensitiveString :: Eq CaseInsensitiveString where
+  eq (CaseInsensitiveString s1) (CaseInsensitiveString s2) =
+    toLower s1 == toLower s2
+
+instance ordCaseInsensitiveString :: Ord CaseInsensitiveString where
+  compare (CaseInsensitiveString s1) (CaseInsensitiveString s2) =
+    compare (toLower s1) (toLower s2)
+
+instance showCaseInsensitiveString :: Show CaseInsensitiveString where
+  show (CaseInsensitiveString s) = "(CaseInsensitiveString " <> s <> ")"
```

Comment:
Just a thought: this (and several other `Show` instances) could use `wrap` now, which would make things a bit shorter.